File: repos_estudiantes_ALVAREZ_segunda_entrega/AGUILAR_MACHICADO_NILS_REINER/PRACTICA 2/COMPOSICION-AGREGACION/Ejercicio5.java
import java.util.ArrayList;

class Jugador {
    protected String nombre;
    protected int numero;
    protected String posicion;

    public Jugador(String nombre, int numero, String posicion) {
        this.nombre = nombre;
        this.numero = numero;
        this.posicion = posicion;
    }

    public String mostrarInfo() {
        return "Jugador: " + nombre + " (#" + numero + ") - Posición: " + posicion;
    }
}

class Portero extends Jugador {
    private String habilidadEspecial;

    public Portero(String nombre, int numero, String habilidadEspecial) {
        super(nombre, numero, "Portero");
        this.habilidadEspecial = habilidadEspecial;
    }

    public String mostrarInfo() {
        return super.mostrarInfo() + ", Habilidad: " + habilidadEspecial;
    }
}

class Defensa extends Jugador {
    private String habilidadEspecial;

    public Defensa(String nombre, int numero, String habilidadEspecial) {
        super(nombre, numero, "Defensa");
        this.habilidadEspecial = habilidadEspecial;
    }

    public String mostrarInfo() {
        return super.mostrarInfo() + ", Habilidad: " + habilidadEspecial;
    }
}

class Mediocampista extends Jugador {
    private String habilidadEspecial;

    public Mediocampista(String nombre, int numero, String habilidadEspecial) {
        super(nombre, numero, "Mediocampista");
        this.habilidadEspecial = habilidadEspecial;
    }

    public String mostrarInfo() {
        return super.mostrarInfo() + ", Habilidad: " + habilidadEspecial;
    }
}

class Delantero extends Jugador {
    private String habilidadEspecial;

    public Delantero(String nombre, int numero, String habilidadEspecial) {
        super(nombre, numero, "Delantero");
        this.habilidadEspecial = habilidadEspecial;
    }

    public String mostrarInfo() {
        return super.mostrarInfo() + ", Habilidad: " + habilidadEspecial;
    }
}

class Equipo {
    private String nombre;
    private ArrayList<Jugador> jugadores;

    public Equipo(String nombre) {
        this.nombre = nombre;
        this.jugadores = new ArrayList<>();
    }

    public void agregarJugador(Jugador jugador) {
        jugadores.add(jugador);
    }

    public void mostrarEquipo() {
        System.out.println("Equipo: " + nombre);
        for (Jugador j : jugadores) {
            System.out.println(j.mostrarInfo());
        }
    }
}
// SIN GETTERS NI SETTERS
class TestEquipo {
    public static void main(String[] args) {
        Equipo equipo = new Equipo("Tigres FC");
        equipo.agregarJugador(new Portero("Carlos", 1, "Atajadas"));
        equipo.agregarJugador(new Defensa("Luis", 4, "Marcaje"));
        equipo.agregarJugador(new Mediocampista("Mario", 8, "Pases"));
        equipo.agregarJugador(new Delantero("Jorge", 9, "Goles"));

        equipo.mostrarEquipo();
    }
}

File: repos_estudiantes_ALVAREZ_segunda_entrega/CAGUAYA_APAZA_YONATHAN_TONY/Practica 2/Composicion-Agregacion/ejercicio5/ejercicio5.py
# 5. Desarrolla un POO para un equipo de fútbol y sus jugadores.
# El equipo está compuesto por jugadores, y si el equipo se destruye, los jugadores también se destruyen.
# Además, los jugadores pueden ser de diferentes tipos (portero, defensa, mediocampista, delantero).
#
# Clase Padre: Jugador<nombre, número, posición>
# Métodos: mostrar_info() (muestra el nombre, número y posición del jugador)
#
# Clases Derivadas: Portero, Defensa, Mediocampista, Delantero (heredan de Jugador)
# Atributos adicionales: habilidad_especial
#
# Clase: Equipo<nombre, jugadores (lista de objetos de tipo Jugador)>
# Métodos: mostrar_equipo() (muestra el nombre del equipo y la información de todos los jugadores)

# DIAGRAMA DE CLASES ERRONEO
class Jugador:
    def __init__(self, nombre, numero, posicion):
        self._nombre = nombre
        self._numero = numero
        self._posicion = posicion

    def mostrar_info(self):
        return f"{self._nombre} - #{self._numero} - {self._posicion}"


class Portero(Jugador):
    def __init__(self, nombre, numero, habilidad_especial):
        super().__init__(nombre, numero, "Portero")
        self._habilidad_especial = habilidad_especial

    def mostrar_info(self):
        return super().mostrar_info() + f" - Habilidad: {self._habilidad_especial}"


class Defensa(Jugador):
    def __init__(self, nombre, numero, habilidad_especial):
        super().__init__(nombre, numero, "Defensa")
        self._habilidad_especial = habilidad_especial

    def mostrar_info(self):
        return super().mostrar_info() + f" - Habilidad: {self._habilidad_especial}"


class Mediocampista(Jugador):
    def __init__(self, nombre, numero, habilidad_especial):
        super().__init__(nombre, numero, "Mediocampista")
        self._habilidad_especial = habilidad_especial

    def mostrar_info(self):
        return super().mostrar_info() + f" - Habilidad: {self._habilidad_especial}"


class Delantero(Jugador):
    def __init__(self, nombre, numero, habilidad_especial):
        super().__init__(nombre, numero, "Delantero")
        self._habilidad_especial = habilidad_especial

    def mostrar_info(self):
        return super().mostrar_info() + f" - Habilidad: {self._habilidad_especial}"


class Equipo:
    def __init__(self, nombre):
        self.__nombre = nombre
        self.__jugadores = []
        self.__crear_jugadores()

    def __crear_jugadores(self):
        self.__jugadores.append(Portero("Carlos", 1, "Atajadas"))
        self.__jugadores.append(Defensa("Luis", 3, "Marcaje"))
        self.__jugadores.append(Mediocampista("Andrés", 8, "Pases"))
        self.__jugadores.append(Delantero("Miguel", 9, "Goles"))

    def mostrar_equipo(self):
        print(f"Equipo: {self.__nombre}")
        for jugador in self.__jugadores:
            print(jugador.mostrar_info())

equipo = Equipo("Tigres del Sur")
equipo.mostrar_equipo()
